### PR TITLE
add postComment function and tests

### PR DESCRIPTION
**Pull request by valy4**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -176,3 +176,77 @@ describe("6.GET /api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No Body");
+      });
+  });
+  test("responds with statusCode 400 if the user sends non-valid username", () => {
```
Comment (robert112233) [inline __tests__/app.test.js]: This test should be a 404. The username is in the correct string format but it is not found in the database. Well done for including it though! 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -176,3 +176,77 @@ describe("6.GET /api/articles/:article_id/comments", () => {
...
+      body: "hdgjhkhk",
+    };
+    return request(app)
+      .post("/api/articles/33003/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
```
Comment (robert112233) [inline __tests__/app.test.js:271]: The rest of your tests are great!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -176,3 +176,77 @@ describe("6.GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+describe.only("7.POST/api/articles/:article_id/comments", () => {
```
Comment (robert112233) [inline __tests__/app.test.js]: Remove .only!

**Review by robert112233 — CHANGES_REQUESTED** [review]
Good work 👍  See comments and submit another PR for review

File: __tests__/app.test.js [hunk 4]
```diff
@@ -176,3 +176,77 @@ describe("6.GET /api/articles/:article_id/comments", () => {
...
+          created_at: expect.any(String),
+          author: "butter_bridge",
+          body: "hjhgfhjf",
+        });
+      });
+  });
+  test("responds with statusCode 400 if the user sends no body", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "",
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Here you still include the body, even it's an empty string. The case for a 400 would be if the 'body' property was missing e. g. `{username: "butter_bridge"}`. Please change this test.

Comment (B7Bprog) [inline __tests__/app.test.js]: Please write another test to make sure that unnecessary properties are ignored, for example when a new comment has an extra property: `{username: "butter_bridge, body: "A brand new comment.", isBestComment: "true"}`.

Your code seems fine selecting only the right properties, but this test should be included.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Valentina, please review my comments and amend your code, then send another pr for us.

**Pull request merged**